### PR TITLE
Feature/svcs 1506 add the task schedules to the sdk

### PR DESCRIPTION
**Pull request by chad-pelly**

### CODE REVIEW

File: src/services/tasks/schedules/types.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+   * **Global Permissions:**
+   * - `CREATE_TASK_SCHEDULE` - Allows a user to create schedules
+   *
+   * @param schedule - The data used to create the schedule
+   * @param options - Additional options for the request
+   */
+  create<T>(
+    schedule: ScheduleCreation<T>,
+    options?: OptionsBase
+  ): Promise<ScheduleCreation<T>>;
```
Comment (chad-pelly) [inline src/services/tasks/schedules/types.ts:23]: This response should actually return of type `Schedule` sadly the service is incorrect and returns `ScheduleCreation` thus, we do the same in the SDK.

A ticket has been made to address the service: https://extrahorizon.atlassian.net/browse/SVCS-1834

File: src/services/tasks/schedules/types.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  creationTimestamp: Date;
+  /** The timestamp at which the schedule will next execute */
+  nextTimestamp: Date;
+  /** The data to be provided to the Function, the type may be user defined */
+  data?: T;
+  /** The priority of the schedule (Higher priorities shall execute first) */
+  priority?: number;
+}
+
+export interface ScheduleCreation<T = Record<string, any>> {
```
Comment (chad-pelly) [inline src/services/tasks/schedules/types.ts:102]: Extract this please, these interfaces share mostly the same properties

File: tests/unit/tasks/schedules/delete.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+describe('Tasks - Schedules - DELETE', () => {
+  const host = 'https://api.xxx.extrahorizon.com';
+  const exh = createClient({
+    host,
+    clientId: '',
+  });
+
+  it('Deletes a Schedule', async () => {
+    const scheduleId = '64ad169e4d29c1b246ee25df';
```
Comment (chad-pelly) [inline tests/unit/tasks/schedules/delete.ts:14]: Randomise

File: src/http/interceptors.ts [hunk 4]
```diff
@@ -54,6 +54,7 @@ const mapDateValues = (value, key) => {
       'creationTimestamp',
       'expiryTimestamp',
       'expireTimestamp',
+      'nextTimestamp',
```
Comment (chad-pelly) [inline src/http/interceptors.ts:57]: Make a ticket to refactor this

**Review by wwaaijer-exh — APPROVED** [review]

**Pull request merged**